### PR TITLE
HPCC-15672 Do not force CPACK_RPM_PACKAGE_ARCHITECTURE

### PR DESCRIPTION
**Pull request by mckellyln**
@Michael-Gardner and @xwang2713 please review.
Can a test on a 32bit x86 system verify the setting is still i386 ?

Signed-off-by: Mark Kelly mark.kelly@lexisnexis.com

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -206,11 +206,11 @@ if(TOP_LEVEL_PROJECT)
     set(CPACK_PACKAGE_VENDOR "HPCC Systems®")
     set(CPACK_PACKAGE_VENDOR_WITHOUT_TRADEMARK "HPCC Systems")
 
-    if(${ARCH64BIT} EQUAL 1)
-        set(CPACK_RPM_PACKAGE_ARCHITECTURE "x86_64")
-    else(${ARCH64BIT} EQUAL 1)
+    # let CPACK_RPM_PACKAGE_ARCHITECTURE get set automatically (uname -m)
+    # except for x86 32bit
+    if("${CPACK_RPM_PACKAGE_ARCHITECTURE}" STREQUAL "i686")
```
Comment (richardkchapman) [inline CMakeLists.txt]: Do you need a check for "not 64-bit" here?

**Comment by xwang2713** [comment]
Looks good

File: CMakeLists.txt (hunk 1 above)
Comment (mckellyln) [inline CMakeLists.txt]: I do not think it is required.

**Comment by mckellyln** [comment]
Final version - left logic unchanged for Windows and for others, set CPACK_RPM_PACKAGE_ARCHITECTURE equal to CMAKE_SYSTEM_PROCESSOR, but then override to i386 only if it was i686 to match previous naming.
@xwang2713 built on all our current platforms (including Windows) and it was correct.
Also tested correctly on Power8 RHEL7
@richardkchapman ready for final review/merge.

**Pull request merged**